woocommerce/woocommerce-android
# Order details view addons button

Summary
==========
Fixes issue #4409 by defining a button inside `OrderDetailsFragment` for every product containing at least one Addon ordered. In order to make the button visible correctly, we have to do a quick check if there's addon presence at the Order by doing a tiny attribute <-> addon name matching in a way that doesn't impact performance.

Screenshots
==========
| Before  | After |
| ------------- | ------------- |
| ![Screenshot_20210817_230901](https://user-images.githubusercontent.com/5920403/129825392-920b3a27-c8ae-43be-9326-c58931aae25d.png) | ![Screenshot_20210817_221933](https://user-images.githubusercontent.com/5920403/129825373-4307b8b5-3080-42f1-8e4d-7190b5aec795.png) |

How to Test
==========
Prerequisites: Have your site with the add ons plugin installed, configured, and with at least one product with add-ons and at least one order of the configured product.

1 - Launch the Woo app.
2 - Navigate to an order with a Product configured with Add-ons, verify if the `View Add-ons` Button is showing up for products that contain Add-ons inside the current Order, but doesn't show up for Products without Add-ons for that order or Products that don't have Add-ons configured at all.
3 - Verify if the `View add-ons` button is visible but unresponsive from click interactions. (You can test if the click trigger is working by placing a Log or breakpoint at Line 464 of `OrderDetailViewModel`)


Update release notes:
